### PR TITLE
Add pre-commit CI build

### PR DESCRIPTION
**Pull request by jrbourbeau**
This PR moves our `black` and `flake8` checks to a separate `pre-commit` CI build on GitHub actions. This is nice because it means there's a single source of truth about which version of `black`/`flake8` we use in CI: our `.pre-commit-config.yaml` file. It also ensures that the pre-commit hooks used locally match up with the checks run in CI.

~~I've also pinned `black` to a specific version (20.8b1) instead of using the latest release on PyPI. This let's us be more intentional as to when we bump our `black` version.~~

- [ ] Tests added / passed
- [ ] Passes `black dask` / `flake8 dask`

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,11 +1,12 @@
 repos:
 -   repo: https://github.com/psf/black
-    rev: stable
+    rev: 20.8b1
     hooks:
     - id: black
-      language_version: python3.7
+      language_version: python3
+      exclude: versioneer.py
```
Comment (jsignell) [inline .pre-commit-config.yaml:7]: Do we need the `exclude` here in addition to the `setup.cfg`?

Comment (jsignell) [reply .pre-commit-config.yaml:7]: oh I see. That is for flake8 only.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,11 +1,12 @@
 repos:
 -   repo: https://github.com/psf/black
-    rev: stable
+    rev: 20.8b1
```
Comment (jsignell) [inline .pre-commit-config.yaml:3]: I think letting black float is fine unless it's been a big source of pain in the past?

**Comment by jsignell** [comment]
I really like this idea of having a single source of truth

**Review by TomAugspurger — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jrbourbeau) [reply .pre-commit-config.yaml:3]: Historically it's been a slight pain point, with a mismatch between local `black` and `black` in CI. Though it sounds like that may have been related to your changes in https://github.com/dask/dask/pull/6641

**Comment by jrbourbeau** [comment]
Thanks for reviewing @jsignell @TomAugspurger!

File: .pre-commit-config.yaml (hunk 2 above)
Comment (jsignell) [reply .pre-commit-config.yaml]: right but there would not longer be a mismatch right?

Comment (jsignell) [reply .pre-commit-config.yaml]: :facepalm: I see now that you changed it. Thanks!

**Pull request merged**